runtime/caml/domain.h

```diff
@@ -97,6 +97,14 @@ Caml_inline intnat caml_domain_alone(void)
   return atomic_load_acquire(&caml_num_domains_running) == 1;
 }
 
+/* The index of the current domain. It is an integer between [0] and
+   [caml_num_domains_running] unique among currently-running domains.
```

Comment:
I suppose you meant to write
```suggestion
   [Max_domains] unique among currently-running domains.
```
since `caml_num_domains_running` is, well, the number of currently running domains, i.e. the number of valid entries in `all_domains[]`.

Reply:
I want to say something more precise than Max_domains (and also, we don't expose the existence of Max_domains to users, which I think is the better choice as it would tie us to an implementation decision that could become a limitation in the future). But you are correct that `caml_num_domains_running` is not quite right. The index of a domain D is in the interval `[0; n]` where `n` was number of running domains *at the time D was created*.

(I am trying to be precise about how "dense" the indexing is, it is more dense than just [0; Max_domains], but also not set in stone implementation details about how indices are chosen at domain creation time.)

Reply:
Here is something more precise than Max_domains and more correct than what I wrote:

> It is an integer unique among
   currently-running domains, in the interval [0; N] where N is the
   peak number of domains running simultaneously so far.

Reply:
Sounds good.